corehq/pillows/case_search.py

```diff
@@ -79,16 +82,26 @@ def transform_case_for_elasticsearch(doc_dict):
     return doc
 
 
+def _format_property(key, value, case_id):
+    if not isinstance(value, str):
+        value = str(value)
```

Comment:
This seems fine for the bug you are fixing, but I think it might be better to fix this at the source - shouldn't all case properties always be strings? If so, I think this belongs in the form processing code when cases are saved, as I think this is likely an assumption we make elsewhere too.

Reply:
@proteusvacuum agreed. There is a discussion about the best approach for that in the Jira ticket.

Reply:
To add to what @snopoke said, a form processor fix is definitely the right solution to this problem, but I think surfacing these incorrect case properties without breaking case search while we determine and implement the right solution (which could be quite complex), is a good way to limit customer pain in the meantime. For now these cases just aren't making it into the index.

Reply:
Great!